vault/auth.go

```diff
@@ -148,6 +155,63 @@ func (v *Vault) UserPassLogin() string {
 	return secret.Auth.ClientToken
 }
 
+// EC2Login - AWS EC2 auth backend
+func (v *Vault) EC2Login() string {
+	if env.Getenv("VAULT_AUTH_AWS_METHOD", "ec2") != "ec2" {
+		return ""
+	}
+
+	if skip := env.Getenv("VAULT_AUTH_AWS_EC2_SKIP", ""); skip != "" {
+		enabled, err := strconv.ParseBool(skip)
+		if err != nil {
+			logFatal("Invalid VAULT_AUTH_AWS_EC2_SKIP value", err)
+		}
+		if enabled {
+			return ""
+		}
+	}
+
+	role := env.Getenv("VAULT_AUTH_AWS_ROLE")
+	mount := env.Getenv("VAULT_AUTH_AWS_MOUNT", "aws")
+
+	vars := map[string]interface{}{}
+
+	if role != "" {
+		vars["role"] = role
+	}
+
+	opts := aws.ClientOptions{}
+
+	timeout := os.Getenv("AWS_TIMEOUT")
+	if timeout != "" {
+		t, err := strconv.Atoi(timeout)
+		if err != nil {
+			logFatal("Invalid AWS_TIMEOUT value '%s' - must be an integer\n", timeout)
+		}
+
+		opts.Timeout = time.Duration(t) * time.Millisecond
```

Comment:
I just exposed a new `MustAtoi` in `github.com/hairyhenderson/gomplate/typeconv`, so you can do this instead:

```go
opts.Timeout = time.Duration(typeconv.MustAtoi(os.Getenv("AWS_TIMEOUT"))) * time.Millisecond
```